### ISSUE TITLE
does it can run on intel cpu or gpu?

### ISSUE THREAD

**Issue by ziyanxzy** [aihacker111/Efficient-Live-Portrait#17]
hi , i see you can run on mac m1 cpu, but it can run on intel cpu or not?

**Comment by aihacker111** [comment]
All CPU, onnx model can run , TensorRT model need GPU @ziyanxzy

**Comment by ziyanxzy** [comment]
intel arc770 etc can we use intel gpu?

**Comment by aihacker111** [comment]
@ziyanxzy you can test and if have any bug , please let me know

**Comment by ziyanxzy** [comment]
how can i set to run on cpu?

**Comment by ziyanxzy** [comment]
because i find it always run on tensorrt instead of cpu

**Comment by aihacker111** [comment]
@ziyanxzy you have to remove flag --run_time, it'll run by onnx model

**Comment by ziyanxzy** [comment]
![image](https://github.com/user-attachments/assets/0fa31644-b94a-483e-a153-ebfadc69dd57)
i do that but still can not run on cpu

**Comment by aihacker111** [comment]
This is run with onnxruntime-gpu, if you want to cpu only, you have to install requirements-cpu.txt

**Comment by ziyanxzy** [comment]
yes i install requirement-cpu,but onnxruntime-gpu in the requirements
![image](https://github.com/user-attachments/assets/dabb4880-343c-450e-bb97-6c7c635ef161)

**Comment by aihacker111** [comment]
@ziyanxzy so sorry, just remove it and replace onnxruntime only , I'll update later

**Comment by ziyanxzy** [comment]
if i uninstall onnxruntime, so i guess it can not run on cpu without any code change?
![image](https://github.com/user-attachments/assets/7b77cdf3-a70e-4b1f-be47-63c0e8c732f9)

**Comment by aihacker111** [comment]
@ziyanxzy onnxruntime is for CPU and onnxruntime-gpu is for GPU
You need to pip uninstall onnxruntime-gpu and install again by pip install onnxruntime (CPU only)

**Comment by ziyanxzy** [comment]
oh thank you ~, but where is the output video? i can not found the output..

**Comment by aihacker111** [comment]
@ziyanxzy it's in animations folder , it's auto create it

**Comment by ziyanxzy** [comment]
yes, i  use this cmd, but can not create a animations folder. python run_live_portrait.py --driving_video "/home/sas/Desktop/valencia/Efficient-Live-Portrait/experiment_examples/examples/driving/d0.mp4" --source_image "/home/sas/Desktop/valencia/Efficient-Live-Portrait/experiment_examples/examples/examples/source/s9.jpg"  --task ['video']

**Comment by aihacker111** [comment]
@ziyanxzy Have you check in the source code, it auto generate in this, also I'm update new feature and fixing the requirements.txt file, you can git clone again

**Comment by aihacker111** [comment]
@ziyanxzy Our not have issues with it , please capture your tree project for me

**Comment by ziyanxzy** [comment]
![image](https://github.com/user-attachments/assets/c657a4d6-8a85-4a0b-93fc-5f9898504a1f)
still can not generate i use example driving video and image

**Comment by aihacker111** [comment]
@ziyanxzy Please capture your inside Efficient-Live-Portrait folder, if not you need to check here for make sure the path is where 
<img width="727" alt="Screenshot 2024-07-30 at 20 51 50" src="https://github.com/user-attachments/assets/a430bb07-085e-4270-a423-668f0e3c9e5c">
If you run it succesfully, it'll print you the path to save output

**Comment by aihacker111** [comment]
<img width="762" alt="Screenshot 2024-07-30 at 20 53 12" src="https://github.com/user-attachments/assets/25c10a96-f815-4db8-b4a9-67fe6aa56fa6">
<img width="359" alt="Screenshot 2024-07-30 at 20 53 44" src="https://github.com/user-attachments/assets/f1b48bb8-f75c-4e6f-8515-5b0eeb15d814">

I'm running on colab or local not have issues

**Comment by ziyanxzy** [comment]
![image](https://github.com/user-attachments/assets/19c2c69f-3b9d-4a9c-9752-31abc0687c90)

**Comment by aihacker111** [comment]
show me the log when finished running

**Comment by aihacker111** [comment]
@ziyanxzy --task not is a list, it's a option image, video or webcam , --task image not --task ['image']

**Comment by ziyanxzy** [comment]
ok i see. i see the code and fint out the task is not a list but a str. now it succeed!!!thank you very much. but i think it will be a little  easy to confuse, because your example is a list. ^^

**Comment by aihacker111** [comment]
@ziyanxzy If you run with tensorrt, please make sure your latest ubuntu and package, and it cannot run on window with TensorRT . Noted this my friend

**Comment by ziyanxzy** [comment]
it doesnt matter~would be run on windows in the future? , and i also want to try run on igpu (intel), and deployment

**Comment by aihacker111** [comment]
@ziyanxzy If you have window os, Please help me build plugin for TensorRT again on window and pull request . Thank you

**Comment by aihacker111** [comment]
@ziyanxzy BTW, I don't like window more haha

**Comment by ziyanxzy** [comment]
if i use tensorrt it will run on cuda, but i want to run on windowscpu or igpu in my pc. i use cpu to run i think it takes not a long time. haha.
i also didnt like windows!

**Comment by ziyanxzy** [comment]
let me try on windows cpu, i think if we want to run on igpu on windows, it may need to convert model to openvino. i will try windows cpu first and give you speed runtime. ^^ thank you . btw , why the onnx run on cpu much faster than python run on cpu. i try to run pytorch model on cpu, but it takes a long time!!!

**Comment by aihacker111** [comment]
@ziyanxzy because onnx is built to run on cpu with back-end api is C++ , and pytorch is not , it’ll take all core of your CPU to run, onnx just take half core

**Comment by ziyanxzy** [comment]
i think  the inference time on cpu depend on the cpu, because i run on a better cpu ,it takes about 4min49s, but the worse one , it takes about 12min25s
